packages/angular_devkit/build_angular/src/angular-cli-files/plugins/single-test-transform.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { logging } from '@angular-devkit/core';
+import { getOptions } from 'loader-utils';
+import { loader } from 'webpack';
+
+export interface SingleTestTransformLoaderOptions {
+  files: string[];
+  logger: logging.Logger;
+}
+
+export default function loader(this: loader.LoaderContext, source: string) {
```

Comment:
Can you add an explanation of what this loader does in a comment?

Something like:
```
/**
 * This loader transforms the default test file to only run tests
 * for some specs instead of all specs.
 * It works by replacing the known content of the auto-generated test file:
 *   const context = require.context('./', true, /\.spec\.ts$/);
 *   context.keys().map(context);
 * with:
 *   const context = { keys: () => ({ map: (_a) => { } }) };
 *   context.keys().map(context);
 * So that it does nothing.
 * Then it also adds import statements for each file in the files options
 * array to import them directly, and thus run the tests there.
 */
```